multiversx/mx-chain-go
# `/transaction/cost` fix endpoint

## Reasoning behind the pull request
- The endpoint `/transaction/cost/` needs to be fixed in order to accurately compute the cost of a provided transaction in gas units. Currently, there are issues with the computation process, resulting in incorrect cost values. The fix aims to address these issues and ensure that the gas cost of the transaction is calculated accurately.
## Proposed changes
- In this PR the response of the endpoint `/transaction/cost/` is extended to include the generated logs from the simulation of the transaction. Previously, the endpoint only provided the computed cost in gas units. However, to enhance transparency and debugging capabilities, it was deemed necessary to include the simulation logs as part of the response.

## Testing procedure
- Call this endpoint with different types of transactions like move balance transactions, ESDTs transactions, and smart contract calls and check if the cost from the response is the same with the `gasUsed` field from the transactions that are indexed in the Elasticsearch database.
## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
